test/lib/ufe/testContextOps.py

```diff
@@ -62,6 +64,26 @@ def reset(self):
         self.addNotif = 0
         self.deleteNotif = 0
 
+class TestNodeMessageHandler:
+    def __init__(self, node):
+        self.nodeDirtyCbId = om.MNodeMessage.addNodeDirtyPlugCallback(node, self.nodeDirty, None)
+        self.nodeDirtyReceived = {}
+        self.attributeChangedCbId = om.MNodeMessage.addAttributeChangedCallback(node, self.attributeChanged, None)
+        self.attributeChangedReceived = {}
+
+    def terminate(self):
+        om.MMessage.removeCallback(self.nodeDirtyCbId)
+        om.MMessage.removeCallback(self.attributeChangedCbId)
+
+    def nodeDirty(self, node, plug, listeners):
+        name = plug.partialName(False, False, False, False, False, True)
+        self.nodeDirtyReceived.setdefault(name, [0,])[0] += 1
```

Comment:
That some bit-obscure Python trick. defaultdict would be clearer and avoid those [0] all over, of good old if/else :)

https://docs.python.org/3/library/collections.html#collections.defaultdict